tests/e2e/testcases/node/monitoring/suite.go

```diff
@@ -0,0 +1,208 @@
+
+package root
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/user"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/ava-labs/avalanche-cli/pkg/ansible"
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/tests/e2e/commands"
+	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"golang.org/x/exp/slices"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	avalanchegoVersion = "v1.10.18"
+	network            = "fuji"
+	networkCapitalized = "Fuji"
+	numNodes           = 2
+	relativePath       = "nodes"
+)
+
+var (
+	hostName         string
+	NodeID           string
+	monitoringHostID string
+	createdHosts     []*models.Host
+	// host names without docker prefix
+	createdHostsFormatted []string
+)
+
+var _ = ginkgo.Describe("[Node monitoring]", func() {
+	ginkgo.It("can create a node", func() {
+		output := commands.NodeCreateWithMonitoring(network, avalanchegoVersion, numNodes)
+		fmt.Println(output)
+		gomega.Expect(output).To(gomega.ContainSubstring("AvalancheGo and Avalanche-CLI installed and node(s) are bootstrapping!"))
+		// parse hostName
+		re := regexp.MustCompile(`Generated staking keys for host (\S+)\[NodeID-(\S+)\]`)
+		match := re.FindStringSubmatch(output)
+		if len(match) >= 3 {
+			hostName = match[1]
+			NodeID = fmt.Sprintf("NodeID-%s", match[2])
+		} else {
+			ginkgo.Fail("failed to parse hostName and NodeID")
+		}
+	})
+	ginkgo.It("creates cluster config", func() {
+		usr, err := user.Current()
+		gomega.Expect(err).Should(gomega.BeNil())
+		homeDir := usr.HomeDir
+		content, err := os.ReadFile(filepath.Join(homeDir, constants.BaseDirName, relativePath, constants.ClustersConfigFileName))
+		gomega.Expect(err).Should(gomega.BeNil())
+		clustersConfig := models.ClustersConfig{}
+		err = json.Unmarshal(content, &clustersConfig)
+		gomega.Expect(err).Should(gomega.BeNil())
+		gomega.Expect(clustersConfig.Clusters).To(gomega.HaveLen(1))
+		gomega.Expect(clustersConfig.Clusters[constants.E2EClusterName].Network.Kind.String()).To(gomega.Equal(networkCapitalized))
+		gomega.Expect(clustersConfig.Clusters[constants.E2EClusterName].Nodes).To(gomega.HaveLen(numNodes))
+		monitoringHostID = clustersConfig.Clusters[constants.E2EClusterName].MonitoringInstance
+		createdHostsFormatted = append(createdHostsFormatted, clustersConfig.Clusters[constants.E2EClusterName].Nodes...)
+	})
+	ginkgo.It("checks prometheus config in monitoring host", func() {
+		usr, err := user.Current()
+		gomega.Expect(err).Should(gomega.BeNil())
+		homeDir := usr.HomeDir
+		monitoringHost, err := ansible.GetInventoryFromAnsibleInventoryFile(filepath.Join(homeDir, constants.BaseDirName, relativePath, constants.AnsibleInventoryDir, "e2e", "monitoring"))
+		gomega.Expect(err).Should(gomega.BeNil())
+		err = commands.DownloadPrometheusConfig(monitoringHost[0], filepath.Join(homeDir, constants.BaseDirName, relativePath, monitoringHostID))
```

Comment:
you need to check that len(monitoringHosts) is 1

Reply:
you need to check that ip address of the host returned is in the  range

Reply:
you need to check that prometheus is installed. ideally that it's running, but installed is sufficient I guess.
I suggest looking into what monitoring intsall script is doing and checking essential elements like creating users installing binaries pkgs and stuff